fantasy-docs.md

```diff
@@ -308,35 +305,69 @@ Every KCL function body is a single expression. If a function body gets really b
 
 ```kcl
 let
-  can_radius = (10, Cm)
-  can_height = (1, Ft)
+  can_radius = Cm(10)
+  can_height = Ft(1)
 in can_of_beans(can_radius, can_height)
 ```
 
 The constants you create in `let` are scoped to the let-in expression. The value of the expression is the `in` part. Let-in blocks are a standard piece of notation from functional programming languages (e.g. in [Elm][elm-let-in], [OCaml][ocaml-let-in] and [Haskell][haskell-let-in]) that help make large expressions more readable. We find they make large KCL functions readable too.
 
-### Constants
+#### Constants
 
 KCL doesn't have any mutation or changes, so there aren't any variables. Files contain a number of functions -- that's it. 
 
-Other languages have named constants and variables. KCL doesn't have variables (because the language describes unchanging geometry and physical characteristics of real-world objects). But it _does_ have named constants. Here's two different ways to declare it:
+Other languages have named constants and variables. KCL doesn't have variables (because the language describes unchanging geometry and physical characteristics of real-world objects). But it _does_ have named constants. Here's how you declare them.
 
 ```kcl
-my_can = can_of_beans((10, Cm), (1, Ft))
+my_can = can_of_beans(Cm(10), Ft(1))
 ```
 
 This declares a named constant called `my_can`, which is the result of calling the `can_of_beans` function we defined above. KCL compiler inferred the type, but you can add a type annotation if you want to:
 
 ```kcl
-my_can: Solid3d = can_of_beans((10, Cm), (1, Ft))
+my_can: Solid3d = can_of_beans(Cm(10), Ft(1))
 ```
 
 This named constant is actually just syntactic sugar for a function that takes 0 parameters. After all, functions called with the same inputs always return the same value -- they're fully deterministic. So a function with 0 parameters is just a function that always returns a constant value. Or, to simplify: it _is_ a constant value. 
 
 Without the syntactic sugar, `my_can` could be declared like this:
 
 ```kcl
-my_can() -> Solid3d = can_of_beans((10, Cm), (1, Ft))
+my_can = (-> Solid3d) => can_of_beans(Cm(10), Ft(1))
+```
+
+Note the function signature. Function signatures are always (parameters -> return type), but here we have no parameters, so the function signature just omits them.
+
+#### Functions as values
+
+Sometimes, functions are parameters to other functions.
+
+```kcl
+doubleDistance = (d: Distance) =>
+    d * 2
+
+doubleAllDistances = (distances: List Distance -> List Distance) =>
+    List.map doubleDistance distances
```

Comment:
I was expecting this syntax: `List.map(doubleDistance, distances)` or `List.map((x) => x * 2, distances)`, based on the example `extrude_closed(material.aluminium, height, circle(radius))`. Are you able to call a function in two ways, with and without parentheses/commas?
1. `extrude_closed(material.aluminium, height, circle(radius))`
2. `extrude_closed material.aluminium height circle(radius)`

Reply:
Oops, that's my bad -- I got confused and thought I was writing Haskell/Elm! Good call out.